### PR TITLE
AAP-79746 fix: use basename for version extraction in ah_collection

### LINKED ISSUE #59: version is extracted incorrectly if path contains "-" characters
##### SUMMARY

If a tar.gz file is provided that adheres to the standard <namespace>-<collection>-<version>.tar.gz naming convention, ah_collection.py extracts the version incorrectly if "-" characters are present in the full path.

For example: `/path/to/my-collection/my_ns-my_collection-1.0.0.tar.gz`
```python
        version = "-".join(path.split("-")[2:]).replace('.tar.gz', '')
```
will result in `my_collection-1.0.0` in stead of `1.0.0`

Using only the basename will solve this issue
```python
from os.path import basename
        version = "-".join(basename(path).split("-")[2:]).replace('.tar.gz', '')
```
will result in `1.0.0`

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

ah_collection.py

##### ANSIBLE VERSION

```paste below
ansible-galaxy [core 2.20.5]
  config file = None
  configured module search path = ['/home/aap-admin/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/aap/aap-admin/ac2/lib64/python3.12/site-packages/ansible
  ansible collection location = /home/aap-admin/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/aap/aap-admin/ac2/bin/ansible-galaxy
  python version = 3.12.12 (main, Apr 16 2026, 00:00:00) [GCC 14.3.1 20250617 (Red Hat 14.3.1-2)] (/opt/aap/aap-admin/ac2/bin/python3)
  jinja version = 3.1.6
  pyyaml version = 6.0.3 (with libyaml v0.2.5)

```

##### COLLECTION VERSION

```paste below

# /home/aap-admin/.ansible/collections/ansible_collections
Collection  Version
----------- -------
ansible.hub 1.0.6

```

##### CONFIGURATION

```paste below
CONFIG_FILE() = None

GALAXY_SERVERS:
```

##### STEPS TO REPRODUCE

below is a snippet from an otherwise working variables file that is used with the roles:
- infra.aap_configuration.hub_namespace
- infra.aap_configuration.hub_collection_repository
- infra.aap_configuration.hub_collection

```yaml
hub_namespaces:
  - name: my_ns
    description: Primary namespace for all of my Ansible collections
    state: present
hub_collection_repositories:
  - name: "my_repo"
    description: "Repository with my collections"
    pulp_labels:
      pipeline: "none"
    distribution:
      name: "my_repo"
      state: present
hub_collections:
  - name: 'my_collection'
    namespace: 'my_ns'
    path: "/home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz"
    version: "1.0.0"
    repository: "my_repo"
    auto_approve: true
    state: present
```

##### EXPECTED RESULTS

I expect the playbook to finish succesfully after creating a Namespace called `my_ns`, which contains a collection called `my_collection` which is connected to repository `my_repo`

##### ACTUAL RESULTS

The namespace, repository and collection are all created and the collection is connected to the correct repository. However, the playbook ends with an error:
```
TASK [infra.aap_configuration.collect_async_status : Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creati
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (50 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (49 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (48 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (47 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (46 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (45 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (44 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (43 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (42 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (41 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (40 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (39 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (38 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (37 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (36 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (35 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (34 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (33 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (32 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (31 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (30 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (29 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (28 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (27 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (26 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (25 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (24 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (23 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (22 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (21 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (20 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (19 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (18 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (17 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (16 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (15 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (14 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (13 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (12 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (11 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (10 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (9 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (8 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (7 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (6 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (5 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (4 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (3 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (2 retries left).
FAILED - RETRYING: [localhost]: Create/Update Collection /home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz | Wait for finish the Collection creation (1 retries left).
fatal: [localhost]: FAILED! =>
    ansible_job_id: j349073181432.76885
    attempts: 50
    changed: false
    finished: 0
    results_file: /home/ansible-tbu/.ansible_async/j349073181432.76885
    started: 1
    stderr: ''
    stderr_lines: <omitted>
    stdout: ''
    stdout_lines: <omitted>
...ignoring

TASK [infra.aap_configuration.collect_async_status : Register value] **************************************************************************************************************
ok: [localhost]

TASK [infra.aap_configuration.collect_async_status : Handle Async Job Failure] ****************************************************************************************************
included: /admin/lib/ansible/ansible_collections/infra/aap_configuration/roles/collect_async_status/tasks/handle_error.yml for localhost => (item=(censored due to no_log))

TASK [infra.aap_configuration.collect_async_status : handle_error | Show error and stop execution] ********************************************************************************
fatal: [localhost]: FAILED! =>
    changed: false
    msg: 'error: Unknown Error'
```

After looking in the logfiles of the HUB API with `podman logs automation-hub-api` I noticed the following:
```
('pulp [bc576906c49445e094741653869c115b]: 127.0.0.1 - - [15/May/2026:07:48:02 +0000] "GET /api/galaxy/pulp/api/v3/content/ansible/collection_versions/?namespace=my_ns&name=my_collection&version=my_collection-1.0.0 HTTP/1.0" 200 52 "-" "Python-urllib/3.9"',)
```

As you can see, the version that is searched for is `my_collection-1.0.0` in stead of `1.0.0` and of course this version does not exist.

When I manually go to the same URL but change the version to `1.0.0` it returns the correct information

### PR DESCRIPTION
**Pull request by Funi1234**
The version extraction split the full file path on hyphens instead of just the filename, producing incorrect versions when directory components contained hyphens. Extract a helper function and operate on the basename.

Issue: AAP-79746

##### SUMMARY

When the path to a collection tarball contains hyphens in directory components (e.g. `/home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz`), the
version was extracted as `my_collection-1.0.0` instead of `1.0.0`. This caused the async upload polling to look for a non-existent version,
eventually timing out with `Unknown Error`.

The fix uses `pathlib.Path(path).name` to extract the basename before splitting on hyphens. The logic is extracted into a standalone `extract_version_from_path()` function with unit tests.

Fixes #59

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
ah_collection

##### ADDITIONAL INFORMATION

Before:
```python
version = "-".join(path.split("-")[2:]).replace('.tar.gz', '')
# "/home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz" -> "my_collection-1.0.0"

After:
version = extract_version_from_path(path)
# "/home/aap-admin/src/my_ns-my_collection-1.0.0.tar.gz" -> "1.0.0"

### CODE REVIEW

**Comment by bmclaughlin** [comment]
## Code Review: PR #64 — AAP-79746 fix: use basename for version extraction in ah_collection

**Commit:** 324f49e | **Date:** 2026-07-16

---

### Verdict: READY_FOR_HUMAN_REVIEW

### Scores
| Lens | Score | Findings |
|---|---|---|
| Functionality | 10/10 | 0 scored findings |
| Security | 10/10 | 0 scored findings (2 pre-existing noted) |
| Quality | 9/10 | 2 Minor (1.0 pts), 2 Nit (0 pts) |
| **Overall** | **9.7/10** | |

### Findings

#### Critical
(none)

#### Major
(none)

#### Minor

1. **[Quality]** **[tests/units/test_ah_collection_version.py:25-26]** Test uses hardcoded relative path for module import; breaks if pytest not run from repo root
   - **Evidence**: `spec = importlib.util.spec_from_file_location("ah_collection", "plugins/modules/ah_collection.py", submodule_search_locations=[])`
   - **Confidence**: MEDIUM
   - **Fix**: Use `pathlib.Path(__file__).resolve().parents[2] / "plugins" / "modules" / "ah_collection.py"` to construct the path relative to the test file rather than the working directory.
   - **Points**: 0.5

2. **[Quality]** **[changelogs/fragments/]** Missing changelog fragment for user-visible bugfix
   - **Evidence**: `changelogs/fragments/` directory is empty. Prior releases (e.g., v1.0.5, v1.0.7) include `Bugfixes` entries for similar fixes. This project uses `antsibull-changelog` with fragment files.
   - **Confidence**: HIGH
   - **Fix**: Add `changelogs/fragments/fix-version-extraction-hyphen-path.yml`:
     ```yaml
     bugfixes:
       - ah_collection - Fix version extraction from collection tarball paths containing hyphens in directory components (https://redhat.atlassian.net/browse/AAP-79746).
     ```
   - **Points**: 0.5

#### Nit

3. **[Quality]** **[tests/units/test_ah_collection_version.py:36-53]** No negative/edge-case test for filenames with hyphenated namespace or collection names
   - **Evidence**: All 8 test cases use underscored namespace/collection names. The function's correctness depends on the Galaxy naming constraint (no hyphens in namespace/name), but no test documents this assumption.
   - **Confidence**: MEDIUM
   - **Points**: 0

4. **[Quality]** **[tests/units/test_ah_collection_version.py:12-20]** Module-level mock injection persists for entire test session
   - **Evidence**: `sys.modules.setdefault(mock_module, MagicMock())` injects 8 mock modules at import time with no cleanup.
   - **Confidence**: LOW
   - **Points**: 0

### Needs Human Judgment

1. **Jira AAP-79746 not accessible** — Cannot verify acceptance criteria or additional context. A human reviewer should confirm the fix matches the ticket's requirements.

2. **Pre-existing version-override behavior** (ah_collection.py:141) — `if auto_approve and path:` unconditionally overwrites any user-supplied `version`. The comment says "if one is not defined" but the guard doesn't check `not version`. This is NOT introduced by this PR but is worth a follow-up ticket. The fix would be: `if auto_approve and path and not version:`.

3. **Changelog fragment** — Whether one is required before merge depends on the team's release process. Some teams add fragments at release-prep time rather than per-PR.

### Verification Results
| Check | Command | Output | Result |
|-------|---------|--------|--------|
| `pathlib` import exists | Read ah_collection.py:103 | `import pathlib` | PASS |
| Function placed at module scope | Read ah_collection.py:108 (PR branch) | `def extract_version_from_path(path):` at module level, before `main()` | PASS |
| Old inline code replaced | Diff line 142 | `-  version = "-".join(path.split("-")[2:]).replace('.tar.gz', '')` replaced by `+  version = extract_version_from_path(path)` | PASS |
| Test covers the exact bug scenario | test_ah_collection_version.py:40 | `("/home/my-user/collections/my_ns-my_collection-1.0.0.tar.gz", "1.0.0")` | PASS |
| Pre-release version handled | test_ah_collection_version.py:43 | `("/tmp/my_ns-my_collection-2.1.0-beta.1.tar.gz", "2.1.0-beta.1")` | PASS |

### Observations

- The fix is correct and minimal. `pathlib.Path(path).name` strips directory components before the hyphen split, which is the root cause fix.
- The function preserves exact pre-existing behavior for well-formed filenames — this is a pure refactor with a targeted fix, not a behavioral change.
- The test suite is well-structured with descriptive parametrize IDs covering 8 scenarios including the exact bug case.
- Security review confirmed no new attack surface — the change is security-neutral to security-positive (narrows input surface by using basename).

### Incidental Findings (out of scope — not scored)

1. **[Functionality]** **[ah_collection.py:141]** Pre-existing: `if auto_approve and path:` overwrites user-supplied `version` unconditionally. If a user passes `version=2.0.0` and `path=/tmp/ns-name-1.0.0.tar.gz`, the module silently uses `1.0.0`. This existed before this PR and is not a regression.

2. **[Security]** **[ah_collection.py:147]** Pre-existing: extracted version is interpolated into API endpoint URL without sanitization. Mitigated by operator-controlled input and server-side validation. The PR improves this by using basename, reducing directory path influence.

### Path to 10/10

1. **[Quality] tests/units/test_ah_collection_version.py:25** (+0.5 points)
   > Use `pathlib.Path(__file__).resolve().parents[2] / "plugins" / "modules" / "ah_collection.py"` instead of relative CWD path.

2. **[Quality] changelogs/fragments/** (+0.5 points)
   > Add a bugfixes changelog fragment for this fix.

### Limitations
This review was performed by an AI agent. LOW-confidence findings and "Needs Human Judgment" items require human verification. This is a first pass, not a final approval.

### Final Score
**Overall: 9.7/10 — READY_FOR_HUMAN_REVIEW**

File: plugins/modules/ah_collection.py [hunk 1]
```diff
@@ -139,7 +143,7 @@ def main():
 
     # approval needs a version, if one is not defined find it from the filename.
     if auto_approve and path:
```
Comment (dragid10) [inline plugins/modules/ah_collection.py]: 🤖

**Medium Care Amount:** The comment above says "if one is not defined find it from the filename," but this condition does not check whether `version` was already provided by the user. A user-supplied `version` parameter gets unconditionally overwritten when `auto_approve` and `path` are both set.

This is what the reporter in [#59](https://github.com/ansible-collections/ansible_hub/issues/59) hit. Their playbook explicitly passes `version: "1.0.0"`, but the module ignored it and extracted from the path instead. Your basename fix resolves their specific case, but the overwrite would still be a problem if someone passes a version that intentionally differs from the filename (e.g., a custom build tag).

Have you considered adding `and not version` to this guard?

```suggestion
    if auto_approve and path and not version:
```

File: plugins/modules/ah_collection.py [hunk 2]
```diff
@@ -105,6 +105,10 @@
 from ..module_utils.ah_module import AHModule
 
 
+def extract_version_from_path(path):
```
Comment (dragid10) [inline plugins/modules/ah_collection.py:108]: 🤖

**Medium-High Care Amount:** Now that this is a named, testable function, it would be good to define what happens with malformed input. If the filename has fewer than two hyphens (e.g., `mycollection.tar.gz` or `name-1.0.0.tar.gz`), `split("-")[2:]` returns an empty list and `"-".join([])` silently returns `""`. That empty string then gets interpolated into the API URL as `collections/{ns}/{name}/versions/`, which either 404s or hits an unrelated endpoint, and the user sees a confusing 50-retry timeout.

Have you considered adding a guard that raises a clear error? For example:

```python
def extract_version_from_path(path):
    parts = pathlib.Path(path).name.replace(".tar.gz", "").split("-")
    if len(parts) < 3:
        raise ValueError(
            "Could not extract version from '{0}'; expected format: "
            "<namespace>-<name>-<version>.tar.gz".format(path)
        )
    return "-".join(parts[2:])
```

The caller could then catch that and call `module.fail_json()` with a helpful message. This would save users from a confusing timeout when the real problem is a misnamed tarball.

File: tests/units/test_ah_collection_version.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    "ah_collection", "plugins/modules/ah_collection.py", submodule_search_locations=[]
+)
+mod = importlib.util.module_from_spec(spec)
+mod.__package__ = "plugins.modules"
+spec.loader.exec_module(mod)
+
+extract_version_from_path = mod.extract_version_from_path
+
+
+@pytest.mark.parametrize(
```
Comment (dragid10) [inline tests/units/test_ah_collection_version.py:61]: 🤖

**Medium Care Amount:** Nice coverage of the happy path! I would love to see a few negative/edge cases added to pin down the function's behavior on bad input. Some ideas:

- `"mycollection.tar.gz"` (no hyphens). Currently returns `""`
- `"name-1.0.0.tar.gz"` (one hyphen, missing namespace). Currently returns `""`
- `"ns-name-1.0.0.tgz"` (wrong extension). Currently returns `"1.0.0.tgz"`
- `"ns-name-1.0.0"` (no extension). Currently returns `"1.0.0"`
- `""` (empty string)

Whether these should assert the current behavior or assert that a `ValueError` is raised depends on whether you add input validation (see my other comment). Either way, documenting the contract for these cases in the test suite would be valuable.

**Review by dragid10 — CHANGES_REQUESTED** [review]
🤖

Thanks for picking this up, Dylan! The root cause analysis is spot-on, and `pathlib.Path(path).name` is the right approach. The parametrized tests with descriptive IDs are also really nice.

Building on Brian's review, a few items I'd like to see addressed before merging:

1. **The `if auto_approve and path:` guard overwrites user-supplied versions** (Medium Care Amount). The reporter in #59 explicitly passed `version: "1.0.0"` but it was still overwritten. Adding `and not version` to the condition would fix this. See inline comment.
2. **Malformed filenames silently produce an empty string** (Medium-High Care Amount). Filenames with fewer than 2 hyphens return `""`, which gets interpolated into the API URL and causes confusing timeouts. Now that this is an extracted function, input validation would be a natural addition. See inline comment.
3. **Negative/edge-case tests** (Medium Care Amount). Would love to see coverage for malformed filenames, wrong extensions, and empty strings. See inline comment.
4. **Missing changelog fragment** (Medium Care Amount). As Brian noted, this needs a `changelogs/fragments/` entry for the bugfix. Something like:

```yaml
bugfixes:
  - ah_collection - Fix version extraction from collection tarball paths containing hyphens in directory components (https://github.com/ansible-collections/ansible_hub/issues/59).
```

**Review by dragid10 — APPROVED** [review]

**Review by bmclaughlin — APPROVED** [review]
